electrode-ota-server-model-acquisition/src/acquisition.js

```diff
@@ -39,25 +42,34 @@ export default (options, dao, weighted, _download, manifest, logger) => {
                         shouldRunBinaryVersion: false
                     };
                 }
-
-                pkg = await dao.getNewestApplicablePackage(params.deploymentKey, params.tags, params.appVersion);
+                // would the client ever send in a range as version?
+                const paramAppVersion = version.coerce(params.appVersion, true).toString();
+                // First try to getNewestApplicablePackage(), matches the incoming appVersion/tag to any of the latest release
+                //  Note that now supporting semver in --targetBinaryVersion, this may not yield desired result
+                pkg = await dao.getNewestApplicablePackage(params.deploymentKey, params.tags, paramAppVersion);
                 if (!pkg) {
-                    // no package match, use latest version that matches tag
+                    // Second try to getNewestApplicablePackage(), use latest version that matches tag
                     pkg = await dao.getNewestApplicablePackage(params.deploymentKey, params.tags);
-                    if (!pkg) {
+                    // don't coerce the targetVersion, it'll remove the range
+                    pkgAppVersion = pkg.appVersion;
+                    // though we get a result, still better to validate against the semver(--targetBinaryVersion)
+                    isTargetBinaryVersion = version.satisfies(paramAppVersion, pkg.appVersion);
+                    if (!pkg || !isTargetBinaryVersion) {
```

Comment:
The first try on `dao.getNewestApplicablePackage()` doesn't yield any results if the appVersion is a semver(due to the reason that query will fail).
The second try is implemented without looking for the version and it provides with the result. So applying the `semver.satisfies()` on the result makes sense and don't proceed if it's not a match.

@datvong-wm, kindly comment on this pls..

Reply:
reverted this implementation as it's no more needed